### PR TITLE
Missing getAttributes documentation

### PR DESCRIPTION
**Pull request by ollieread**
This PR adds documentation for `getAttributes` methods, following on from #1539.

This covers;

- `ReflectionClassConstant::getAttributes`
- `ReflectionFunctionAbstract::getAttributes`
- `ReflectionParameter::getAttributes`
- `ReflectionProperty::getAttributes`

### CODE REVIEW

File: reference/reflection/reflectionclassconstant/getattributes.xml [hunk 1]
```diff
@@ -25,15 +25,25 @@
...
+      Flags for determining how to filter the results, if <parameter>name</parameter>
+      is provided.
+     </para>
+     <para>
+      Default is <literal>0</literal> which will only return results for attributes that
+      are of the class <parameter>name</parameter>.
+     </para>
+     <para>
+      The only other option available, is to use <constant>ReflectionAttribute::IS_INSTANCEOF</constant>,
+      which will instead use <literal>instanceof</literal> for filtering.
```
[the remark this answers is not in the document]
Comment (ollieread) [reply reference/reflection/reflectionclassconstant/getattributes.xml]: Good shout @cmb69, added.

File: reference/reflection/reflectionclassconstant/getattributes.xml [hunk 2]
```diff
@@ -47,6 +33,125 @@
...
  </refsect1>
 
+ <refsect1 role="examples">
+  &reftitle.examples;
+  <para>
+   <example>
+    <title>Basic usage</title>
+    <programlisting role="php">
+     <![CDATA[
+<?php
```
Comment (Girgias) [inline reference/reflection/reflectionclassconstant/getattributes.xml:43]:
```suggestion
<![CDATA[
<?php
```
And everywhere else

Comment (ollieread) [reply reference/reflection/reflectionclassconstant/getattributes.xml:43]: That's my bad, I fixed that in my previous pr, no idea how that's made it's way back in. Will get sorted, thanks.

**Comment by cmb69** [comment]
This is marked as draft. Please switch to "ready for review" when the PR is ready to be reviewed. :)

**Comment by ollieread** [comment]
> This is marked as draft. Please switch to "ready for review" when the PR is ready to be reviewed. :)

I marked it as draft because I'm trying to think of good examples for properties and parameters. The whole red apple thing worked for all the others but doesn't make sense for those, so I'm reluctant to reuse it.

**Comment by afilina** [comment]
@ollieread Gentle nudge to author. Do you plan to continue working on this, or close if it's no longer relevant?

**Comment by ollieread** [comment]
> @ollieread Gentle nudge to author. Do you plan to continue working on this, or close if it's no longer relevant?

Ah, thank you for the nudge. Will pick it up soon and come up with some good examples.

**Comment by ollieread** [comment]
I've added examples for `ReflectionProperty::getAttributes()` and `ReflectionParameter::getAttributes()`. I'm not _super_ happy with the examples, but any example that would be worth anything would be exceedingly long and most likely include numerous other things. I'm open to suggestions if people aren't happy with the current examples.

Furthermore, I think it may be worth it to expand the "Reflection Book" part of the documentation to include some real-world examples and a bit more of an explanation as to what's going on. This wouldn't be attribute-specific, though a expansion on the attributes guide may also be worth it.

File: reference/reflection/reflectionclassconstant/getattributes.xml [hunk 3]
```diff
@@ -47,6 +33,125 @@
...
+interface Color {
+}
+
+#[Attribute]
+class Fruit {
+}
+
+#[Attribute]
+class Red implements Colour {
```
Comment (cmb69) [inline reference/reflection/reflectionclassconstant/getattributes.xml:127]: Typo: should be "Color" to match the interface name (and yeah, we prefer American English spelling). The same issue 6 times below.

**Review by cmb69 — COMMENTED** [review]
Thank you!

> I'm not _super_ happy with the examples, but any example that would be worth anything would be exceedingly long and most likely include numerous other things.

Actually, I am quite happy with the examples: simple and easy to understand. Fancy stuff shouldn't go into the PHP manual, in my opinion.

Besides the typos (see comment below), this looks great to me!

**Comment by cmb69** [comment]
Thank you!

**Pull request merged**